### DISCUSSION TITLE
bugfix

### DISCUSSION THREAD

**Discussion by SCP1715** [ra3xdh/qucs_s#205] [Show and tell]
I noticed a problem with qucs: when you try to drag a component from the components window to the workspace, the application crashes, although when you click everything works
I solved this problem in the following way:

`class iconCompInfoStruct
{
public:
	int catIdx; // index of the component Category
	int compIdx; // index of the component itself in the Category
	iconCompInfoStruct()
	{
		catIdx = -1;
		compIdx = -1;
		qRegisterMetaTypeStreamOperators<iconCompInfoStruct>("iconCompInfoStruct");
	};
	iconCompInfoStruct(int cat, int comp)
	{
		catIdx = cat;
		compIdx = comp;
		qRegisterMetaTypeStreamOperators<iconCompInfoStruct>("iconCompInfoStruct");
	}
};

QDataStream &operator<<(QDataStream &out, const iconCompInfoStruct &myObj)
{
	out << myObj.catIdx;
	out << myObj.compIdx;
	return out;
}

QDataStream &operator>>(QDataStream &in, iconCompInfoStruct &myObj)
{
	in >> myObj.catIdx;
	in >> myObj.compIdx;
	return in;
}`

**Comment by ra3xdh** [comment]
This may be related to #187 that is already fixed. Only platforms using Qt5.15 were affected. Are you still experience the crash using the latest snapshot from the `current` branch from Git? Are you experiencing crash on every drag'n'drop event? I cannot reproduce the crash on drag'n'drop event on my machine using both Qt5.9 and Qt5.15. What is your platform?

**Reply by SCP1715** [reply]
I tried the solution to the problem described in #187, but the program still crashes
I am using QT5.8.0 and VS 2017
Perhaps my solution is more versatile
